core/src/main/kotlin/com/strumenta/kolasu/codegen/ASTCodeGenerator.kt

```diff
@@ -30,12 +30,15 @@ abstract class ASTCodeGenerator<R : Node> {
     }
 
     fun printToString(ast: R): String {
-        return PrinterOutput(this.nodePrinters, nodePrinterOverrider)
-            .apply { this.print(ast) }
-            .text()
+        return printerOutput().apply { this.print(ast, prefix, postfix) }.text()
     }
 
     fun printToFile(root: R, file: File) {
         file.writeText(printToString(root))
     }
+
+    open fun printerOutput() = PrinterOutput(this.nodePrinters, nodePrinterOverrider)
```

Comment:
Why should someone be able to override this method?

Would not be confusing to set up nodePrinters and then have this method overriden and the nodePrinters being ignored?

Reply:
Because they may want to customize the printer output (changing indentation, newlines, etc.)
Indeed overriding per se is not the only way. We could add a method `customizePrinterOutput`.

Reply:
Would that be a sort of post-processing? If so one could make a `postProcessing` method which by default does nothing and make that one overridable, otherwise I think your approach would be best